xdsl/transforms/individual_rewrite.py

```diff
@@ -44,43 +44,26 @@ def match_and_rewrite(self, op: arith.DivUIOp, rewriter: PatternRewriter) -> Non
             rewriter.replace_matched_op([], [mul_op.lhs])
 
 
-INDIVIDUAL_REWRITE_PATTERNS_BY_OP_CLASS: dict[
-    type[Operation], tuple[RewritePattern, ...]
-] = {
-    arith.AddiOp: (AdditionOfSameVariablesToMultiplyByTwo(),),
-    arith.DivUIOp: (DivisionOfSameVariableToOne(),),
+INDIVIDUAL_REWRITE_PATTERNS_BY_NAME: dict[str, dict[str, RewritePattern]] = {
+    arith.AddiOp.name: {
+        AdditionOfSameVariablesToMultiplyByTwo.__name__: AdditionOfSameVariablesToMultiplyByTwo()
+    },
+    arith.DivUIOp.name: {
+        DivisionOfSameVariableToOne.__name__: DivisionOfSameVariableToOne()
+    },
 }
 """
-Dictionary where the key is an Operation and the value is a tuple of rewrite pattern(s) associated with that operation. These are rewrite patterns defined in this class.
+Extra rewrite patterns available to ApplyIndividualRewritePass
 """
 
-CANONICALIZATION_PATTERNS_BY_OP_CLASS: dict[
-    type[Operation], tuple[RewritePattern, ...]
-] = {
-    op: trait.get_canonicalization_patterns()
-    for dialect in get_all_dialects().values()
-    for op in dialect().operations
-    if (trait := op.get_trait(HasCanonicalizationPatternsTrait)) is not None
-}
-"""
-Dictionary where the key is an Operation and the value is a tuple of rewrite pattern(s) associated with that operation. These are the xdsl canonicalization patterns.
-"""
 
-REWRITE_BY_NAMES: dict[str, dict[str, RewritePattern]] = {
-    op.name: {
-        pattern.__class__.__name__: pattern
-        for pattern in INDIVIDUAL_REWRITE_PATTERNS_BY_OP_CLASS.get(op, ())
-        + CANONICALIZATION_PATTERNS_BY_OP_CLASS.get(op, ())
-    }
-    for op in set(INDIVIDUAL_REWRITE_PATTERNS_BY_OP_CLASS)
-    | set(CANONICALIZATION_PATTERNS_BY_OP_CLASS)
-}
-"""
-Returns a dictionary representing all possible rewrites. Keys are operation names, and
-values are dictionaries. In the inner dictionary, the keys are names of patterns
-associated with each operation, and the values are the corresponding RewritePattern
-instances.
-"""
+def _get_pattern(op: Operation, pattern_name: str) -> RewritePattern | None:
```

Comment:
```suggestion
def _get_canonicalization_pattern(op: Operation, pattern_name: str) -> RewritePattern | None:
```
I thought this was including the individual rewrites at first